### ISSUE TITLE
try and document auth0 integration with other frameworks

### ISSUE THREAD

**Issue by edublancas** [ploomber/doc#214]
We recently shipped an [integration with Auth0](https://docs.cloud.ploomber.io/en/latest/user-guide/password.html#auth0-authentication), we already tested with:

- [streamlit](https://github.com/ploomber/doc/commit/d5311f875ff50160e68a89785b2ce94ad2090fc9)
- [panel](https://github.com/ploomber/doc/commit/dd0be5f765f05540826d96ab7e2bca09823d0433)

but we need to check if the following frameworks work:

- [x] shiny (R) https://github.com/ploomber/doc/pull/225
- [x] Solara https://github.com/ploomber/doc/pull/220
- [x] chainlit https://github.com/ploomber/doc/pull/222
- [x] dash https://github.com/ploomber/doc/pull/222
- [x] voila https://github.com/ploomber/doc/pull/224

if they don't, we just need to report it here and we'll tackle it later. if it works, we need to:

- upload a basic app
    - it should show how to read the request headers that display the user name
    - show how to add the logout button (note that the logout feature is broken currently, we'll push a fix soon, but you can put the logout link now)
    - sample panel app: https://github.com/ploomber/doc/tree/main/examples/panel/app-with-auth0
    - sample streamlit app: https://github.com/ploomber/doc/blob/main/examples/streamlit/app-with-auth0/app.py
- we also need to modify the docs so the framework's section documents the auth0 integration, for an example, look at
    - [streamlit](https://github.com/ploomber/doc/commit/d5311f875ff50160e68a89785b2ce94ad2090fc9)
    - [panel](https://github.com/ploomber/doc/commit/dd0be5f765f05540826d96ab7e2bca09823d0433)

**Comment by edublancas** [comment]
@neelasha23 please do solara, and chainlit (can be a single PR for both)

**Comment by edublancas** [comment]
note: I started working on the Shiny R example but I could not read the [headers](https://docs.cloud.ploomber.io/en/latest/user-guide/password.html#knowing-who-logged-in): https://github.com/ploomber/doc/commit/76b55690a3567b4cc22e24b239c5fbc8c4687f8f (the function returns an empty string)

**Comment by neelasha23** [comment]
i have added example for Solara: https://github.com/ploomber/doc/pull/220

The first time I added Auth0 it worked (the login page came up). But the second time I created a new application in Auth0, and redeployed the project with new Client ID and Base URL, updated the callback URL. But it keeps bypassing the login page. Am I missing something? @edublancas

**Comment by edublancas** [comment]
I'm unsure if this is a solara issue, auth0 or an issue in our end, please open an issue

**Comment by neelasha23** [comment]
Opened an [issue](https://github.com/ploomber/cloud-backend/issues/1008). This doesn't seem to be a problem with the framework because I never saw the login page came up after the first Solara app login.  Have shared a recording with you on Slack.  I have only opened [this PR] since I saw some intermittent redirection to the auth0 domain, and I was also able to access the `X-Auth-Name`. So it looks like auth0 has remembered the first login details @edublancas
